### PR TITLE
Use testing json assertions

### PR DESCRIPTION
**Pull request by frankban**
Big branch (apologies!) but mostly mechanical.

Also updated dependencies.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/602/

File: internal/v4/api_test.go [hunk 1]
```diff
@@ -16,8 +16,8 @@ import (
 	"strconv"
 	"strings"
 	"time"
-
 	jc "github.com/juju/testing/checkers"
+	httptesting "github.com/juju/testing/httptesting"
```
Comment (rogpeppe) [inline internal/v4/api_test.go]: We don't need the alias here, do we?

File: internal/legacy/api_test.go [hunk 2]
```diff
@@ -13,8 +13,8 @@ import (
 	"net/http/httptest"
 	"os"
 	"time"
-
```
Comment (mhilton) [inline internal/legacy/api_test.go:16]: I believe this blank line should still be there.

**Comment by rogpeppe** [comment]
LGTM with one trivial. Run sortimports ./... before submission to make sure all the imports are grouped correctly.

File: internal/v4/log_test.go [hunk 3]
```diff
@@ -8,8 +8,8 @@ import (
 	"encoding/json"
 	"net/http"
 	"time"
-
```
Comment (mhilton) [inline internal/v4/log_test.go:11]: And this one.

File: internal/v4/relations_test.go [hunk 4]
```diff
@@ -6,7 +6,7 @@ package v4_test
 import (
 	"fmt"
 	"net/http"
-
```
Comment (mhilton) [inline internal/v4/relations_test.go:9]: This one too.

File: internal/v4/stats_test.go [hunk 5]
```diff
@@ -9,7 +9,7 @@ import (
 	"net/url"
 	"strings"
 	"time"
-
```
Comment (mhilton) [inline internal/v4/stats_test.go:12]: This one.

**Comment by mhilton** [comment]
:+1: LGTM

**Comment by frankban** [comment]
Thanks to sortimports I've fixed the import grouping errors.
Thnak you both for the reviews!

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/604/

**Comment by frankban** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**